### PR TITLE
Fixes to the Monthly stats report following QA process with TAD

### PR DESCRIPTION
**Pull request by duncanjbrown**
## Context

Before the next Monthly stats report (the "External report") job runs on Monday, we need to incorporate the changes from last month's QA process. This PR does that at the same time as rebuilding the specs so that all reports use the same dataset to report on, which is much more like real life. Once the reports are all properly under test, we can begin to refactor the reports themselves. There's more to do, but this PR is big enough as it is.

## Changes proposed in this pull request

Rebuild each monthly report and its specs one by one. Once everything's working against the specs, refactor.

Also add some Rake tasks for importing and exporting reports, which we needed last month (and might well need this month)

## Guidance to review

It's big, but it should be pretty commit by commit.

For each report fixed, first look at the changes to `MonthlyStatisticsTestHelper`. This is the common test data all report specs use. Then look at the specs, all of which are passing at each step. Then satisfy yourself that the changes to the code aren't unreasonable. The first (`by_status`) report is the biggest change — downhill from there.

(Note that the candidate reports (sex/area/age) got their specs in subsequent commits that I couldn't rebase together without unacceptable pain)

## Link to Trello card

https://trello.com/c/j4kcaT5T/4191-monthly-report-clean-up-code

### CODE REVIEW

File: lib/tasks/monthly_report.rake [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  :monthly_statistics_candidates_by_age_group,
-  :monthly_statistics_candidates_by_area,
+  # :monthly_statistics_applications_by_course_age_group,
+  # :monthly_statistics_applications_by_course_type,
+  # :monthly_statistics_applications_by_primary_specialist_subject,
+  # :monthly_statistics_applications_by_provider_area,
+  # :monthly_statistics_applications_by_secondary_subject,
+  # :monthly_statistics_applications_by_status,
+  # :monthly_statistics_candidates_by_age_group,
+  # :monthly_statistics_candidates_by_area,
```
Comment (stevehook) [inline lib/tasks/monthly_report.rake]: Remove comments?

**Review by stevehook — COMMENTED** [review]
Still working my way through this but I have a question... Does the simplification of the way we handle deferrals (i.e. not more counting from last cycle) mean that the second paragraph of the footnote on the view template is now redundant...

image

**Review by stevehook — APPROVED** [review]
This all looks good to me. (I've just done a visual inspection rather than try to check the numbers as that's been covered by others!). Some really nice improvements and much more consistent.

File: spec/support/monthly_statistics_test_helper.rb [hunk 2]
```diff
@@ -35,7 +35,7 @@ def generate_monthly_statistics_test_data
     form = create(:application_form, phase: 'apply_1')
     create(:application_choice,
            :with_declined_offer,
-           course_option: course_option_with(level: 'secondary', program_type: 'higher_education_programme', region: 'north_west'),
+           course_option: course_option_with(level: 'secondary', program_type: 'higher_education_programme', region: 'north_west', subjects: [create(:subject, name: 'Art and design', code: 'W1'), create(:subject, name: 'History', code: 'V1')]),
```
[the remark this answers is not in the document]
Comment (duncanjbrown) [reply spec/support/monthly_statistics_test_helper.rb:38]: I was a bit lazy! I think it's ok?

**Comment by duncanjbrown** [comment]
> Still working my way through this but I have a question... Does the simplification of the way we handle deferrals (i.e. not more counting from last cycle) mean that the second paragraph of the footnote on the view template is now redundant...
> 
> image

This is a direct consequence of that. That copy isn't great — effectively we're saying we're double counting them as UCAS counted them last year.

**Pull request merged**